deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/AutoPromoter.java

```diff
@@ -122,32 +123,116 @@ boolean isCurrentDeployRetirable(String envId, DeployBean currDeployBean) throws
         return false;
     }
 
-    void promoteBuild(EnvironBean currEnvBean, DeployBean currDeployBean, int size) throws Exception {
+    //This contains the logic about if there is a promote build pending
+    public PromoteResult computePromoteBuildResult(EnvironBean currEnvBean, DeployBean currDeployBean, int size,
+                                                   PromoteBean promoteBean) throws Exception {
+        Date now = new Date();
+        long mustBefore = Long.MAX_VALUE;
+        String schedule = promoteBean.getSchedule();
+        if (!StringUtils.isEmpty(schedule)) {
+            long start_date = 0;
+            if (currDeployBean!=null){
+                start_date = currDeployBean.getStart_date();
+            }
+            else{
+                //Never deploy. We should only check against the time of the latest build. Otherwise
```

Comment:
I am thinking it would be intuitive to use the last update time of schedule itself instead of latest build time to determine whether to trigger the deployment.  If people want to start a deployment immediately, they can always trigger one manually.